### DISCUSSION TITLE
Using napari for GUI ?

### DISCUSSION THREAD

**Discussion by vfdev-5** [Project-MONAI/MONAILabel#80] [General]
I was wondering if it was already considered to using napari tools for UI in the plugins ?

For example, https://github.com/napari/magicgui looks interesting to create widgets from functions...

**Comment by SachidanandAlle** [comment]
Thanks for sharing about napari.  Currently have limited scope for client sided integrations.

We picked 3D Slicer (which is quite famous tool with large set of features, plugins and open-source community) as an example to demonstrate e2e functionality for MONAI Label.  Actually this time we took one step ahead and doing the reference implementation for client-sided plugin.

In later stages, we would like to showcase this as reference to bring in other open source viewers/developers to integrate with MONAILabel.  Please note, MONAILabel doesn't have any UI of it's own.  It provides endpoints for client sided UI applications to integrate along with their existing features/functionalities to make use of AI features provided through MONAILabel.  In future if we are need to create UI widgets as part of MONAILabel, we can definitely evaluate magicgui option (which is abstracted over QT).

Adding @pdogra89

**Comment by sofroniewn** [comment]
Hello, one of the napari maintainers here. If at any point you or anyone else would like any help building a MONAI napari plugin or using magicgui don't hesitate to reach out. I'd be happy to help and collaborate. I'm very excited about the MONAI project in general and would love to see some its functionality made available through napari! Thanks for even thinking of us :-)

**Comment by bhoeckendorf** [comment]
Sorry about being so late to this conversation. I've actually been looking into this, and you can see my efforts [here](https://github.com/bhoeckendorf/napari-monailabel).

Apart from submitting new labels, which isn't implemented yet, everything _is intended to_ work, but I'd consider it experimental. I plan to also add a configurable backend app, to help users get started quickly.

**Reply by sofroniewn** [reply]
Wow @bhoeckendorf that's awesome! @neuromusic might be interested in this too. Do let us know if things aren't working or you have additional feature requests for napari. You can join our [zulip](https://napari.zulipchat.com/login/) to talk with more of the team too.

**Reply by bhoeckendorf** [reply]
Happy if people are excited about this. MONAILabel is fantastic, and I hope the combination with a really flexible & hackable client like napari will be useful for both projects.

Thanks for the support and the link, @sofroniewn. Indeed I might have a few questions, but they are a little more open-ended and I need to consolidate my own thoughts first. I'll reach out when I'm ready.

**Reply by SachidanandAlle** [reply]
Thank you for taking up this challenge.. may be once you are ready.. you can help to create a wiki/demo video for anyone to try it out..
